packages/@react-aria/dnd/src/utils.ts

```diff
@@ -159,12 +161,12 @@ export class DragTypes implements IDragTypes {
     this.includesUnknownTypes = !hasFiles && dataTransfer.types.includes('Files');
   }
 
-  has(type: string) {
-    if (this.includesUnknownTypes || (type === 'directory' && this.types.has(GENERIC_TYPE))) {
+  has(type: string | DirectorySymbolType) {
```

Comment:
does `string | symbol` work here? Might be good to keep this a little more vague rather than only allowing `directorySymbol`, in case we add more special symbol types in the future.